src/MainWindow.vala

```diff
@@ -276,9 +273,10 @@ public class Atlas.MainWindow : Adw.ApplicationWindow {
         } catch (Error error) {
             warning (error.message);
             return;
+        } finally {
+            loc_store.remove_all ();
```

Comment:
I prefer moving this line out of the try-catch (before declaring `places` variable because `loc_store` has nothing to do with `search_async()` and plus this unnecessarily makes the nest deeper.

I guess you respected the current flow and used `finally` but I personally think we should use it only when needed (like freeing up resources used in the `try` block).